/*!
  @mainpage –ü—Ä–∏–ª–æ–∂–µ–Ω–∏–µ –¥–ª—è —Ñ–∏—Ç–Ω–µ—Å –∫–ª—É–±–∞
    ‚ö°–ü—Ä–∏–ª–æ–∂–µ–Ω–∏—è –¥–ª—è –≤–µ–¥–µ–Ω–∏—è –±–∞–∑—ã –¥–∞–Ω–Ω—ã—Ö —Ñ–∏—Ç–Ω–µ—Å –∫–ª—É–±–∞

    üìÑ–ü—Ä–∏–ª–æ–∂–µ–Ω–∏–µ –≤—ã–ø–æ–ª–Ω–µ–Ω–æ —Å –ø–æ–º–æ—â—å—é –≥—Ä–∞—Ñ–∏—á–µ—Å–∫–æ–π –±–∏–±–ª–∏–æ—Ç–µ–∫–∏ Qt –∏ –ø—Ä–æ–≥—Ä–∞–º–º–Ω–æ–π –±–∏–±–ª–∏–æ—Ç–µ–∫–∏ SQLite

    üî®–û—Å–Ω–æ–≤–Ω—ã–µ —Ñ—É–Ω–∫—Ü–∏–∏:

        - –î–æ–±–∞–≤–ª–µ–Ω–∏–µ/—Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞–Ω–∏–µ/—É–¥–∞–ª–µ–Ω–∏–µ –∫–ª–∏–µ–Ω—Ç–æ–≤

        - –î–æ–±–∞–≤–ª–µ–Ω–∏–µ/—Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞–Ω–∏–µ/—É–¥–∞–ª–µ–Ω–∏–µ —Ç–∞—Ä–∏—Ñ–æ–≤

        - –î–æ–±–∞–≤–ª–µ–Ω–∏–µ/—É–¥–∞–ª–µ–Ω–∏–µ —É—Å–ª—É–≥

    üîß–í –ø–∞–ø–∫–µ "default db" –ª–µ–∂–∏—Ç —Ç–µ—Å—Ç–æ–≤–∞—è –±–∞–∑–∞ –¥–∞–Ω–Ω—ã—Ö (—á–∞—Å—Ç–∏—á–Ω–æ –∑–∞–ø–æ–ª–Ω–µ–Ω–Ω–∞—è), —á—Ç–æ–±—ã –∞–∫—Ç–∏–≤–∏—Ä–æ–≤–∞—Ç—å –µ—ë,
        –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ —Å–∫–æ–ø–∏—Ä–æ–≤–∞—Ç—å/–ø–µ—Ä–µ–Ω–µ—Å—Ç–∏ –µ—ë –≤ –ø–∞–ø–∫—É —Å –±–∏–ª–¥–æ–º –ø—Ä–æ–≥—Ä–∞–º–º—ã

*/
#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include "database.h"
#include "editwindow.h"

QT_BEGIN_NAMESPACE
namespace Ui { class MainWindow; }
QT_END_NAMESPACE

/*!
    \brief –ö–ª–∞—Å—Å –æ–ø–∏—Å—ã–≤–∞–µ—Ç –ª–æ–≥–∏–∫—É —Ä–∞–±–æ—Ç—ã –æ—Å–Ω–æ–≤–Ω–æ–≥–æ –æ–∫–Ω–∞

    –û—Å–Ω–æ–≤–Ω–æ–µ –æ–∫–Ω–æ —Å–æ–¥–µ—Ä–∂–∏—Ç 2 –≤–∫–ª–∞–¥–∫–∏: –¥–ª—è —Ä–∞–±–æ—Ç—ã —Å –∫–ª–∏–µ–Ω—Ç–∞–º–∏ –∏ –¥–ª—è —Ä–∞–±–æ—Ç—ã —Å —É—Å–ª—É–≥–∞–º–∏/—Ç–∞—Ä–∏—Ñ–∞–º–∏.

*/

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:

/*!
    –ö–æ–Ω—Å—Ç—Ä—É–∫—Ç–æ—Ä –ø—Ä–∏–Ω–∏–º–∞–µ—Ç —Ä–æ–¥–∏—Ç–µ–ª—å—Å–∫–∏–π –∫–ª–∞—Å—Å.
*/

    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –∑–∞–ø–æ–ª–Ω–µ–Ω–∏—è —Ç–∞–±–ª–∏—Ü—ã –∏–º–µ–Ω–∞–º–∏ –∫–ª–∏–µ–Ω—Ç–æ–≤.
*/

    void parse_db();

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –æ—á–∏—Å—Ç–∫–∏ –±–ª–æ–∫–∞ "–ò–Ω—Ñ–æ—Ä–º–∞—Ü–∏—è –æ —Ç–∞—Ä–∏—Ñ–µ".
*/

    void clear_info();

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –∑–∞–ø–æ–ª–Ω–µ–Ω–∏—è –±–ª–æ–∫–∞ "–ò–Ω—Ñ–æ—Ä–º–∞—Ü–∏—è –æ —Ç–∞—Ä–∏—Ñ–µ".

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    row - –Ω–æ–º–µ—Ä —Å—Ç—Ä–æ–∫–∏ —Å –∫–ª–∏–µ–Ω—Ç–æ–º, –¥–ª—è –∫–æ—Ç–æ—Ä–æ–≥–æ –Ω—É–∂–Ω–æ –≤—ã–≤–µ—Å—Ç–∏ –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏—é.
*/

    void parse_info(int row);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –≤—ã–∫–ª—é—á–µ–Ω–∏—è/–≤–∫–ª—é—á–µ–Ω–∏—è –∫–Ω–æ–ø–æ–∫ —É–¥–∞–ª–µ–Ω–∏—è/–∏–∑–º–µ–Ω–µ–Ω–∏—è –∫–ª–∏–µ–Ω—Ç–∞.

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    pos - true (–≤–∫–ª—é—á–∏—Ç—å) / false (–≤—ã–∫–ª—é—á–∏—Ç—å).
*/

    void enabled_b(bool pos);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –≤—ã–≤–æ–¥–∞ –≤—Å–µ—Ö —Ç–∞—Ä–∏—Ñ–æ–≤ –≤ —Ç–∞–±–ª–∏—Ü—É.
*/

    void parse_tariff();

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –∑–∞–ø–æ–ª–Ω–µ–Ω–∏—è –±–ª–æ–∫–∞ "–ò–Ω—Ñ–æ—Ä–º–∞—Ü–∏—è –æ —Ç–∞—Ä–∏—Ñ–µ".

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    row - –Ω–æ–º–µ—Ä —Å—Ç—Ä–æ–∫–∏ —Å —Ç–∞—Ä–∏—Ñ–æ–º, –¥–ª—è –∫–æ—Ç–æ—Ä–æ–≥–æ –Ω—É–∂–Ω–æ –≤—ã–≤–µ—Å—Ç–∏ –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏—é.
*/

    void parse_tariff(int row);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –æ—á–∏—Å—Ç–∫–∏ –±–ª–æ–∫–∞ "–ò–Ω—Ñ–æ—Ä–º–∞—Ü–∏—è –æ —Ç–∞—Ä–∏—Ñ–µ".
*/

    void clear_info_tariff();
/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –≤—ã–∫–ª—é—á–µ–Ω–∏—è/–≤–∫–ª—é—á–µ–Ω–∏—è –∫–Ω–æ–ø–æ–∫ —É–¥–∞–ª–µ–Ω–∏—è/–∏–∑–º–µ–Ω–µ–Ω–∏—è —Ç–∞—Ä–∏—Ñ–∞.

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    pos - true (–≤–∫–ª—é—á–∏—Ç—å) / false (–≤—ã–∫–ª—é—á–∏—Ç—å).
 */

    void enabled_b_tariff(bool pos);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –∫–æ–Ω—Ç—Ä–æ–ª—è –æ—Ç–æ–±—Ä–∞–∂–µ–Ω–∏—è –±–ª–æ–∫–∞ —Å–º–µ–Ω—ã —Ç–∞—Ä–∏—Ñ–∞.

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    pos - true (–Ω–µ –æ—Ç–æ–±—Ä–∞–∂–∞—Ç—å) / false (–æ—Ç–æ–±—Ä–∞–∂–∞—Ç—å).
 */

    void close_change_tariff(bool pos);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –∫–æ–Ω—Ç—Ä–æ–ª—è –æ—Ç–æ–±—Ä–∞–∂–µ–Ω–∏—è –±–ª–æ–∫–∞ –ø—Ä–æ–¥–ª–µ–Ω–∏—è —Ç–∞—Ä–∏—Ñ–∞.

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    pos - true (–Ω–µ –æ—Ç–æ–±—Ä–∞–∂–∞—Ç—å) / false (–æ—Ç–æ–±—Ä–∞–∂–∞—Ç—å).
 */

    void close_change_date(bool pos);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –ø—Ä–æ–≤–µ—Ä–∫–∏ –Ω–∞ –≤–æ–∑–º–æ–∂–Ω–æ—Å—Ç—å –ø—Ä–æ–¥–ª–µ–Ω–∏—è —Ç–∞—Ä–∏—Ñ–∞.
*/

    void isChanged();

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –∫–æ–Ω—Ç—Ä–æ–ª—è –≤–æ–∑–º–æ–∂–Ω–æ—Å—Ç–∏ —Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞–Ω–∏—è –ø–æ–ª–µ–π "—Ç–µ–ª." –∏ "–§–ò–û".

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    pos - true (–Ω–µ–ª—å–∑—è —Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞—Ç—å) / false (–º–æ–∂–Ω–æ —Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞—Ç—å).
*/

    void FIO_phone_readonly(bool pos);

/*!
    \brief –ú–µ—Ç–æ–¥ –¥–ª—è –ø–æ–∏—Å–∫–∞ –ø–æ –ø–æ–¥—Å—Ç—Ä–æ–∫–µ.

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    arg1 - –ø–æ–¥—Å—Ç—Ä–æ–∫–∞, –ø–æ –∫–æ—Ç–æ—Ä–æ–π –æ—Å—É—â–µ—Å—Ç–≤–ª—è–µ—Ç—Å—è –ø–æ–∏—Å–∫.
*/

    void search(const QString &arg1);


signals:

/*!
    \brief –°–∏–≥–Ω–∞–ª –¥–ª—è –ø–µ—Ä–µ–¥–∞—á–∏ id –æ –≤—ã–±—Ä–∞–Ω–Ω–æ–º —Ç–∞—Ä–∏—Ñ–µ –≤ –æ–∫–Ω–æ –∫–ª–∞—Å—Å–∞ "EditWindow".

    –ü–∞—Ä–∞–º–µ—Ç—Ä—ã:

    id - primary key –¥–ª—è –∏–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏–∏ –∫–æ–Ω–∫—Ä–µ—Ç–Ω–æ–≥–æ —Ç–∞—Ä–∏—Ñ–∞.
 */

    void sentId(QString id);

private slots:

    void on_renew_b_clicked();

    void on_FIO_t_cellClicked(int row);

    void on_add_b_clicked();

    void on_del_b_clicked();

    void on_date_b_clicked();

    void on_FIO_t_cellActivated(int row);

    void on_change_b_clicked();

    void on_tariff_t_cellClicked(int row);

    void on_renew_tariff_clicked();

    void on_tariff_t_cellActivated(int row);

    void on_add_tariff_clicked();

    void on_delete_tariff_clicked();

    void on_edit_tariff_clicked();

    void on_serv_b_clicked();


    void on_ok_t_b_clicked();

    void on_no_t_b_clicked();

    void on_ok_d_b_clicked();

    void on_no_d_b_clicked();

    void on_edit_clicked();

    void on_Search_textChanged(const QString &arg1);

    void on_tabWidget_currentChanged();

private:
    Ui::MainWindow *ui;
    DataBase *db;
    EditWindow *window;
};
#endif // MAINWINDOW_H
